python_modules/dagster/dagster/core/execution/context/output.py

```diff
@@ -307,7 +312,7 @@ def partition_key(self) -> str:
                 "For more details: https://github.com/dagster-io/dagster/issues/7900"
             )
 
-        return self.step_context.partition_key
+        return self._partition_key
```

Comment:
In the main execution case (not `build_output_context`), I'm not seeing how we still get the partition key. Maybe I'm missing something?

Reply:
arf I meant to make this a conditional